pages/task/read/routers/view.js

```diff
@@ -23,22 +23,34 @@ const getRelevantActivity = activityLog => activityLog.filter(log => {
 module.exports = () => {
   const app = Router();
 
-  // if task is a project redirect to the project interface
+  // get relevant versionId if task is for a project.
   app.use((req, res, next) => {
     const model = get(req.task, 'data.model');
+
     if (model === 'project') {
       req.projectId = get(req.task, 'data.id');
       req.establishmentId = get(req.task, 'data.data.establishmentId');
       return req.api(`/establishment/${req.establishmentId}/project/${req.projectId}`)
         .then(({ json: { data } }) => {
-          req.versionId = data.draft.id;
-          res.redirect(req.buildRoute('project.version'));
+          req.project = data;
         })
+        .then(() => next())
         .catch(next);
     }
     next();
   });
 
+  // redirect to project page if ?redirect=true
+  app.get('/', (req, res, next) => {
+    const model = get(req.task, 'data.model');
+    const status = get(req.task, 'status');
+    if (model === 'project' && req.query.redirect === 'true') {
+      req.versionId = status === 'resolved' ? req.project.granted.id : req.project.draft.id;
+      return res.redirect(req.buildRoute('project.version'));
+    }
+    next();
+  });
```

Comment:
This is for the case where they've hit "amend and resubmit" right?

Reply:
yes, then the POST project/grant takes care of the actual resubmitting, however currently the comments box is ignored, we _might_ want to just read that from the session

Reply:
no, actually it is not, sorry I didn't actually look at the code - this actually isn't used anymore